semantics/src/traversal/functions.rs

```diff
@@ -94,19 +94,34 @@ pub fn func_def(
 }
 
 fn validate_all_paths_return_or_revert(
-    body: &[Spanned<fe::FuncStmt>],
+    block: &[Spanned<fe::FuncStmt>],
 ) -> Result<(), SemanticError> {
-    // This will need to become more sophisticated when we introduce branching logic
-    // because we then need to follow different code paths and check that they
-    // all return or revert.
-    for statement in body {
+    for statement in block {
```

Comment:
Looks like this might be better off as a single match statement.

Reply:
The reason I didn't do that is because I would have to add one arm with an empty `{}` which doesn't look very elegant imho. Or did you think of something else?